web/src/app/modules/forms/modules/validation/util/validation-cache.ts

```diff
@@ -0,0 +1,88 @@
+import { ValidationResult } from '../../../../../validation/validation-result';
+import { IContainer } from '../../../../../model/IContainer';
+
+export class ValidationCache {
+    constructor() {
+        this.dataCache = {};
+        this.contentCache = {};
+    }
+
+    private dataCache: {[url: string]: ValidationResult[]};
+    private contentCache: {[url: string]: string};
+
+    public removeEntry(url: string) {
+        if (!(url in this.dataCache)) {
+            return;
+        }
+        let results = this.dataCache[url];
+        delete this.dataCache[url];
+        delete this.contentCache[url];
+        /* Changing one element might make the results of other elements to become invalid as well:
+        * A => Result[A,B], Result[A]
+        * B => Result[A,B]
+        *
+        * delete A
+        * A => []
+        * B => Result[A,B]   <-- Now invalid and needs to be removed
+        */
+        let elements = [].concat(...results.map(res => res.elements));
+        elements.forEach( (element: IContainer) => {
+            delete this.dataCache[element.url];
+            delete this.contentCache[element.url];
+        });
+    }
+
+    private static getContentString(content: string[]) {
+        return content.sort().join(' ');
+    }
+
+    public isCached(url: string, content: string[]) {
+        if (!(url in this.dataCache)) {
+            return false;
+        }
+        let cStr = ValidationCache.getContentString(content);
+        return this.contentCache[url] === cStr;
+    }
+
+    public getEntry(url: string) {
+        return this.dataCache[url];
+    }
+
+    public addEntriesToCache(url: string, content: string[], entries: ValidationResult[]) {
+        if (!(url in this.dataCache)) {
+            this.dataCache[url] = [];
+        }
+        this.dataCache[url] = entries;
+        // One Result might affect multiple elements
+        // These Results are stored in the cache of every individual element
+        entries.forEach( entry => {
+            entry.elements.map(el => el.url).forEach(elUrl => {
+                if (elUrl !== url) {
+                    if (!(elUrl in this.dataCache)) {
+                        this.dataCache[elUrl] = [];
+                    }
+                    this.dataCache[elUrl].push(entry);
+                }
+            });
+        });
+    }
+
+    public findValidationResults(parentElement: IContainer, resultFilter?: (result: ValidationResult) => boolean) {
```

Comment:
Please add a valid return type to this function. Right now, you are just returning an arbitrary object (and treat it like a map). You could use, e.g.: `{ [key: string]: TypeOfFilteredResult }`